### ISSUE TITLE
Minor flaw in typings

### ISSUE THREAD

**Issue by surma** [GoogleChromeLabs/comlink#329]
```ts
const {x} = wrap<{x: SomeType}>(new Worker("./worker.ts"));
```

For the initial destructuring, `x` should be of type `SomeType`, but is currently of type `Promise<SomeType>`.

**Comment by pocesar** [comment]
isn't that expected, since each member is an async getter, regardless of depth? I had no issue when creating a React wrapper for that (as in, the types are correct), for both class instantiation to individual methods

**Comment by pocesar** [comment]
I did found an weird issue though. when there are multiple return values on the function, they get split, and although the promisified result is correct, TS won't allow you to use it, having to resort to typecasting the generated value from `wrap` (in my case, it's a plain, non async function, inside the worker)

![image](https://user-images.githubusercontent.com/461084/67246050-ceb52c00-f433-11e9-9859-30984ef8b694.png)

![image](https://user-images.githubusercontent.com/461084/67246213-366b7700-f434-11e9-81ec-ad87191f54d0.png)

without typecasting, TS gives the following error:

```
This expression is not callable.
  Each member of the union type '(<TResult1 = null, TResult2 = never>(onfulfilled?: ((value: null) => TResult1 | PromiseLike<TResult1>) | null | undefined, onrejected?: ((reason: any) => TResult2 | PromiseLike<...>) | null | undefined) => Promise<...>) | (<TResult1 = string, TResult2 = never>(onfulfilled?: ((value: string) => TResult1 | PromiseLike...' has signatures, but none of those signatures are compatible with each other.
```

which is weird, because both are Promises, although different result types

**Comment by surma** [comment]
Fixed in v4.1.0
